src/Moryx.AbstractionLayer.Products.Endpoints/ProductManagementController.cs

```diff
@@ -0,0 +1,332 @@
+            if (productInstance == null)
+                return NotFound();
+            return _productConverter.ConvertProductInstance(productInstance);
+        }
+       
+        [HttpGet]
+        [Route("instances")]
+        public ProductInstanceModel[] GetInstances([FromQuery] long[] ids)
+        {
+            var instances = _productManagement.GetInstances(ids);
+            var modelList = new List<ProductInstanceModel>();
+            foreach(var instance in instances)
+                modelList.Add(_productConverter.ConvertProductInstance(instance));
+            return modelList.ToArray();
+        }
+
+        [HttpPost]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [Route("instances")]
+        public ActionResult<ProductInstanceModel> CreateInstance(string identifier, short revision, bool save)
+        {
+            var identity = new ProductIdentity(WebUtility.HtmlEncode(identifier), revision);
+            var productType = _productManagement.LoadType(identity);
+            if (productType == null)
+                return BadRequest($"Product type not found");
+            var instance = _productManagement.CreateInstance(productType, save);
+            return _productConverter.ConvertProductInstance(instance);
+        }
+
+        [HttpPost]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [Route("instances")]
+        public ActionResult SaveInstance(ProductInstanceModel instanceModel)
+        {
+            if (instanceModel == null)
+                return BadRequest($"Instance model was empty");
+            var type = ReflectionTool.GetPublicClasses<IProductType>(t => t.Name == instanceModel.Type)
+                    .FirstOrDefault();
+            if (type == null)
+                return NotFound();
+            var productType = (IProductType)Activator.CreateInstance(type);
+            var productInstance = _productConverter.ConvertProductInstanceBack(instanceModel, productType);
+           _productManagement.SaveInstance(productInstance);
+            return Ok();
+        }       
+
+        #endregion
+        #region recipes
+        [HttpGet]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [Route("recipes/{id}")]
+        public ActionResult<RecipeModel> GetRecipe(long id)
+        {
+            if (id == 0)
+                return BadRequest($"Id was 0");
+            var recipe = _productManagement.LoadRecipe(id);
+            if (recipe == null)
+                return NotFound();
+            return ProductConverter.ConvertRecipe(recipe);
+        }
+
+        [HttpGet]
+        [Route("recipes/types")]
+        public RecipeDefinitionModel[] GetRecipeTypes()
```

Comment:
Same as above. If we want those, maybe just use types/definitions and recipes/definitions